src/amo/components/OverallRating.js

```diff
@@ -70,25 +97,57 @@ export class OverallRatingBase extends React.Component {
   }
 }
 
-export const mapStateToProps = (state) => ({
-  apiState: state.api,
-  userId: state.auth && state.auth.userId,
-});
+export const mapStateToProps = (state, ownProps) => {
+  const userId = state.auth && state.auth.userId;
+  let userReview;
+
+  // Look for an already saved review by this user for this add-on/version.
+  if (userId && state.reviews) {
+    const allUserReviews = state.reviews[userId];
+    log.info(dedent`Checking state for review by user ${userId},
+      addonId ${ownProps.addonId}, versionId ${ownProps.version.id}`);
+    const addonReviews =
+      allUserReviews ? allUserReviews[ownProps.addonId] : null;
+
+    if (addonReviews) {
+      for (const reviewId of Object.keys(addonReviews)) {
+        const review = addonReviews[reviewId];
+        if (review.isLatest) {
+          userReview = review;
+          log.info('Found the latest review in state for this component',
+                   userReview);
+          break;
+        }
+      }
+    }
+  }
+
+  return {
+    apiState: state.api,
+    userReview,
+    userId,
+  };
+};
 
 export const mapDispatchToProps = (dispatch) => ({
-  createRating({ router, addonSlug, addonId, userId, ...params }) {
+
+  loadSavedRating({ userId, addonId }) {
+    return getUserReviews({ userId, addonId, onlyTheLatest: true })
+      .then((review) => {
+        if (review) {
+          dispatch(setReview(review));
+        } else {
+          log.info(
+            `No saved review found for userId ${userId}, addonId ${addonId}`);
+        }
+      });
+  },
+
+  submitRating({ router, addonSlug, ...params }) {
     return submitReview({ addonSlug, ...params })
       .then((review) => {
         const { lang, clientApp } = params.apiState;
-        // TODO: when we have a user_id in the API response, we
-        // could probably use that instead.
-        // https://github.com/mozilla/addons-server/issues/3672
-        dispatch(setReview({
-          addonId,
-          rating: review.rating,
-          versionId: review.version.id,
-          userId,
-        }));
+        dispatch(setReview(review));
```

Comment:
The API helper prevents changing the version anytime you are editing a review: https://github.com/mozilla/addons-frontend/pull/1374/files#diff-3d585cb73e6d00650d856aaf80203732R13

Without this check, the API returns an error since you can't change the version.